### PR TITLE
#2258. Add more private fields promotion tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Private-fields-promotion/not_promotable_A06_t01.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/// hold:
+/// - It refers to a field (not a getter)
+/// - The field is private
+/// - The field is final
+/// - There is no getter with the same name elsewhere in the library
+/// - There is no non-final field with the same name elsewhere in the library
+/// - There is no implicit noSuchMethod forwarder with the same name elsewhere
+///   in the library.
+///
+/// @description Checks that if there an implicit `noSuchMethod` forwarder with
```
Comment (eernstg) [inline LanguageFeatures/Private-fields-promotion/not_promotable_A06_t01.dart:15]:
```suggestion
/// @description Checks that if there is an implicit `noSuchMethod` forwarder with
```

File: LanguageFeatures/Private-fields-promotion/not_promotable_A06_t02.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/// hold:
+/// - It refers to a field (not a getter)
+/// - The field is private
+/// - The field is final
+/// - There is no getter with the same name elsewhere in the library
+/// - There is no non-final field with the same name elsewhere in the library
+/// - There is no implicit noSuchMethod forwarder with the same name elsewhere
+///   in the library.
+///
+/// @description Checks that if there an implicit `noSuchMethod` forwarder with
```
Comment (eernstg) [inline LanguageFeatures/Private-fields-promotion/not_promotable_A06_t02.dart:15]:
```suggestion
/// @description Checks that if there is an implicit `noSuchMethod` forwarder with
```

File: LanguageFeatures/Private-fields-promotion/promotion_A04_t04.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+/// hold:
+/// - It refers to a field (not a getter)
+/// - The field is private
+/// - The field is final
+/// - There is no getter with the same name elsewhere in the library
+/// - There is no non-final field with the same name elsewhere in the library
+/// - There is no implicit noSuchMethod forwarder with the same name elsewhere
+///   in the library.
+///
+/// @description Checks that if there an implicit `noSuchMethod` forwarder of a
```
Comment (eernstg) [inline LanguageFeatures/Private-fields-promotion/promotion_A04_t04.dart:15]:
```suggestion
/// @description Checks that if there is an implicit `noSuchMethod` forwarder of a
```

**Review by eernstg — COMMENTED** [review]
Looks good, just noted a couple of typos.

**Comment by sgrekhov** [comment]
Typos fixed

**Review by eernstg — APPROVED** [review]
LGTM

**Comment by eernstg** [comment]
Thanks!

**Pull request merged**